src/main/java/juke/command/AddCommand.java

```diff
@@ -0,0 +1,112 @@
+package juke.command;
+
+import juke.exception.JukeInvalidParameterException;
+import juke.exception.JukeMissingArgumentException;
+import juke.exception.JukeParseException;
+import juke.task.Deadline;
+import juke.task.Event;
+import juke.task.TaskType;
+import juke.task.Todo;
+
+public class AddCommand extends Command {
+    private TaskType type;
+    
+    public AddCommand(TaskType type) {
+        this.type = type;
+    }
+    
+    @Override
+    public Command checkParametersAndArguments() {
+        switch (this.type) {
+        case EVENT:
+            if (this.hasParameter("at")) {
+                if (this.hasArgument("at")) {
+                
+                } else {
+                    this.result = Result.error(new JukeMissingArgumentException(this.type.getCommandName()));
+                    return this;
+                }
+            } else {
+                this.result = Result.error(new JukeMissingArgumentException(this.type.getCommandName()));
+                return this;
+            }
```

Comment:
Wouldn't it be better to rearrange the if statement so that 1: it avoids arrowhead coding and 2: there's no blank if statement?
```
            if (!this.hasParameter("at")) {
                this.result = Result.error(new JukeMissingArgumentException(this.type.getCommandName()));
                return this;
            } else if (!this.hasArgument("at")) {
                this.result = Result.error(new JukeMissingArgumentException(this.type.getCommandName()));
                return this;
            }
```